feat(web-analytics): add marketing insights and routing

jabahamondes opened the pull request:
> [!IMPORTANT]
> 👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Problem

Need for marketing analytics

## Changes

- Introduced new marketing tab in web analytics with relevant routing and insights.
- Updated constants and scene configurations to include marketing features.
- Enhanced WebAnalyticsDashboard to conditionally display marketing costs based on feature flags.
- Adjusted logic to handle marketing product tab and its associated breadcrumbs.
- Updated WebPropertyFilters to hide filters when on the marketing tab.

Code is behind a FF

## Did you write or update any docs for this change?

- [ ] I've added or updated the docs
- [ ] I've reached out for help from the docs team
- [x] No docs needed for this change

## How did you test this code?

Manually

<img width="1195" alt="image" src="https://github.com/user-attachments/assets/b06fe713-af0e-4abb-a838-4b6d7afa2003" />

Commit:
feat(web-analytics): add marketing insights and routing

- Introduced new marketing tab in web analytics with relevant routing and insights.
- Updated constants and scene configurations to include marketing features.
- Enhanced WebAnalyticsDashboard to conditionally display marketing costs based on feature flags.
- Adjusted logic to handle marketing product tab and its associated breadcrumbs.
- Updated WebPropertyFilters to hide filters when on the marketing tab.

Commit:
Merge branch 'master' into feat/creating-marketing-template

Commit:
refctor: Reverting temp code

Commit:
Merge branch 'master' into feat/creating-marketing-template

jabahamondes commented on frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx:
```diff
@@ -1221,6 +1232,84 @@ export const webAnalyticsLogic = kea<webAnalyticsLogicType>([
+                                    series: [
+                                        {
+                                            kind: NodeKind.DataWarehouseNode,
+                                            id: 'bigquery.google_ads_test1.google_ads_2',
```
These are just testing tables, but they will change in the future, so it doesn't matter to have them by now

lricoy replied:
Hmm, but do they change anything? I haven't used the DataWarehouseNode option before, will this fail if they don't exist? 

ps: I will try locally and see 😅

jabahamondes replied:
It will fail @lricoy , but it doesn't matter because I'll will remove that. I will do the following:

allowing to add datawarehouse resources and map the schema
iterate on datawarehouse resources -> create that chat based on that
Something like that

robbie-c replied:
Nice, yeah, iterating on the tables and only showing this if it's supported makes sense.

Bonus points is if the fallback component tells the user how to get set up / has a link to the docs / links to the setup page directly.

(You don't need to do all of this in one PR, but we should do it before we have actual users using it)

Commit:
feat(web-analytics): add marketing tab to dashboard with beta label
- Introduced a new marketing tab in the Web Analytics Dashboard, displayed conditionally based on feature flags.
- The tab includes a "Beta" label for clarity.

Commit:
Adding comment

lricoy commented on frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx:
```diff
@@ -1221,6 +1235,99 @@ export const webAnalyticsLogic = kea<webAnalyticsLogicType>([
                     ]
                 }
 
+                if (productTab === ProductTab.MARKETING) {
```
Not related to your PR, but now that we're adding more tabs, we should consider refactoring this main kea logic into more specific ones. There is some shared logic for sure, but 80% is the query payload creation.

lricoy commented on frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx:
```diff
@@ -2290,6 +2397,78 @@ export const webAnalyticsLogic = kea<webAnalyticsLogicType>([
+                        ORDER BY cc.total_cost DESC
+                        LIMIT 20
+                    `,
+                    filters: {
+                        dateRange: {
+                            date_from: dateFilter.dateFrom || '-7d',
+                            date_to: dateFilter.dateTo || 'now()',
+                        },
+                    },
```
Just a question: Does this work, and is it injected into the query, or did the types require it?

jabahamondes replied:
It does not work haha I'll remove it by now, but I will create a query runner eventually for all this logic

lricoy commented on frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx:
```diff
@@ -170,6 +174,8 @@ const loadPriorityMap: Record<TileId, number> = {
     [TileId.PAGE_REPORTS_TIMEZONES]: 13,
     [TileId.PAGE_REPORTS_LANGUAGES]: 14,
     [TileId.PAGE_REPORTS_TOP_EVENTS]: 15,
+    [TileId.MARKETING]: 16,
+    [TileId.MARKETING_CAMPAIGN_BREAKDOWN]: 17,
```
I guess we can break this down by tab now, they only matter for on-screen concurrency. No need to change it, just commenting.

Commit:
fix(web-analytics): update marketing campaign breakdown tile priority and remove date filters
- Changed priority of the MARKETING_CAMPAIGN_BREAKDOWN tile to 1.
- Removed unnecessary date filters from the web analytics logic.

lricoy reviewed (approved):
It is behind a flag, so I guess it is all good, you'll know what cleaning up you'll need to do after :)

robbie-c commented on frontend/src/scenes/web-analytics/WebPropertyFilters.tsx:
```diff
@@ -8,14 +8,18 @@ import { IconWithCount } from 'lib/lemon-ui/icons'
 
     const [displayFilters, setDisplayFilters] = useState(false)
 
+    if (productTab === ProductTab.MARKETING) {
```
Nit: I'd prefer the parent handle this logic, rather than this component rendering null

robbie-c commented on frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx:
```diff
@@ -2290,6 +2399,68 @@ export const webAnalyticsLogic = kea<webAnalyticsLogicType>([
+                        ),
+                        campaign_pageviews AS (
+                            SELECT 
+                                properties.$utm_campaign as campaign_name,
```
I think this shouldn't have the dollar, just `properties.utm_campaign`

robbie-c commented on frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx:
```diff
@@ -2290,6 +2399,68 @@ export const webAnalyticsLogic = kea<webAnalyticsLogicType>([
+                                count(*) as pageviews,
+                                uniq(distinct_id) as unique_visitors
+                            FROM events 
+                            WHERE event = '$pageview' 
```
Nice, it's easy to see how you could extend this to other conversion goals (like how Brian has both pageview and signup here https://docs.google.com/spreadsheets/d/1JxE2t0C6P9s_5Ee_TTsbNrayNAgwx1kjA9jgqdp3dDw/edit?gid=2007488702#gid=2007488702)

Commit:
feat(web-analytics): Conditional rendering of WebPropertyFilters based on productTab

- Added productTab to the state in WebAnalyticsFilters and WebAnalyticsModal.
- Updated WebPropertyFilters to render conditionally based on productTab, excluding it for the MARKETING tab.
- Adjusted campaign costs breakdown query to use the correct property name for campaign.
